tests/admin/test-class-admin-editor-specific-replace-vars.php

```diff
@@ -8,247 +8,250 @@
+
+	/** @var WPSEO_Admin_Editor_Specific_Replace_Vars_Test */
+	protected $class_instance;
+
+	/**
+	 * Set up the class which will be tested.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->class_instance = new WPSEO_Admin_Editor_Specific_Replace_Vars_Double();
+		// add_filter( 'wpseo_editor_specific_replace_vars', array( $this, 'filter_editor_specific_replacevars' ) );
+	}
+
+	/**
+	 * Tests that get_shared_replace_vars removes all replacement variables that occurs in the editor specific
+	 * replacement variables.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::get_shared_replace_vars
+	 */
+	public function test_get_shared_replace_vars_filters_editor_specific_replace_vars() {
+		$replace_vars_list = array(
+			array(
+				'name' => 'searchphrase',
+				'label' => 'Searchphrase',
+				'value' => '',
+			),
+			array(
+				'name' => 'title',
+				'label' => 'title',
+				'value' => '',
+			),
+		);
+
+		$this->assertEquals(
+			array( 'title' ),
+			$this->class_instance->get_generic(
+				$replace_vars_list
+			)
+		);
+	}
+
+	/**
+	 * Tests that determine_for_term can detect a category.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_term
+	 */
+	public function test_determine_for_term_with_a_category() {
+		$this->assertEquals( 'category', $this->class_instance->determine_for_term( 'category' ) );
+	}
+
+	/**
+	 * Tests that determine_for_term can detect a tag.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Var::determine_for_term
+	 */
+	public function test_determine_for_term_with_a_tag() {
+		$this->assertEquals( 'post_tag', $this->class_instance->determine_for_term( 'post_tag' ) );
+	}
+
+	/**
+	 * Tests that determine_for_term can detect a post_format.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_term
+	 */
+	public function test_determine_for_term_with_a_post_format() {
+		$this->assertEquals( 'post_format', $this->class_instance->determine_for_term( 'post_format' ) );
+	}
+
+	/**
+	 * Tests that determine_for_post defaults to post when no actual post variable is passed along.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_without_a_wp_post_instance() {
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post( null ) );
+	}
+
+	/**
+	 * Tests that a homepage is succesfully determined.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::is_homepage
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_homepage() {
+		// Backup the current values for these options.
+		$show_on_front = get_option( 'show_on_front' );
+		$page_on_front = get_option( 'page_on_front' );
+
+		$post = $this->create_and_get_with_post_type( 'page' );
+
+		// Overwrite the options used in the is_homepage function.
+		update_option( 'show_on_front', 'page' );
```

Comment:
I think there is no usecase for a homepage specific situation in `determine_for_post`